File: c/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>
#include <string.h>

int check_letters(string text);
int check_words(string text);
int check_sentence(string text);

int main(void)
{

    string text = get_string("Enter your text: ");

    float letters = check_letters(text);
    float words = check_words(text);
    float sentence = check_sentence(text);
    float avg_L = 0;
    float avg_S = 0;

    avg_L = letters / words * 100;
    avg_S = sentence / words * 100;

    int index = round(0.0588 * avg_L - 0.296 * avg_S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }

}

int check_letters(string text)
{
    int i;
    int letter = 0;
    for (i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letter++;
        }
        else
        {
            continue;
        }
    }

    return letter;

}


int check_words(string text)
{
    int i;
    int word = 0;
    for (i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            word++;
        }
        else
        {
            continue;
        }
    }

    return word + 1;

}


int check_sentence(string text)
{
    int i;
    int sentence = 0;
    for (i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence++;
        }
        else
        {
            continue;
        }
    }

    return sentence;

}

File: phython/sentimental-readability/readability.py
# Import the get_string function from the cs50 library
from cs50 import get_string

# Get a string from the user
text = get_string("Enter your text: ")

# Initialize variables to store the number of letters, sentences, and words in the text
letters = 0
sentence = 0
words = 1

# Get the length of the text
l = len(text)

# Iterate over the characters in the text
for i in range(l):
    # If the character is an alphabetical character, increment the letter count
    if (text[i].isalpha()):
        letters += 1
    # If the character is a space, increment the word count
    if (text[i] == " "):
        words += 1
    # If the character is a period, exclamation mark, or question mark, increment the sentence count
    if text[i] in [".", "!", "?"]:
        sentence += 1

# Calculate the average number of letters per 100 words
avg_L = letters / words * 100

# Calculate the average number of sentences per 100 words
avg_S = sentence / words * 100

# Calculate the readability index using the formula given in the prompt
index = round(0.0588 * avg_L - 0.296 * avg_S - 15.8)

# Print the appropriate message based on the value of the readability index
if (index < 1):
    print("Before Grade 1")
elif (index > 15):
    print("Grade 16+")
else:
    print(f"Grade {index}")
